libraries/wasm-jit/Source/WASM/WASMSerialization.cpp

```diff
@@ -406,8 +406,8 @@ namespace WASM
 
 	struct LocalSet
 	{
-		Uptr num;
-		ValueType type;
+		Uptr num{};
```

Comment:
Stack trace for it
```
WASMSerialization.cpp:28
WASMSerialization.cpp:417
WASMSerialization.cpp:504
```

Starting from top in line `WASMSerialization.cpp:504` we define: `LocalSet localSet` which is unitialized. It is passed down to `serialize(Stream& stream,ValueType& type)` in WASMSerialization.cpp:26 where LocalSet::type is read (uninitialized).

It happens for every usage of `serializeFunctionBody(InputStream& sectionStream,Module& module,FunctionDef& functionDef)` in WASMSerialization.cpp.
Unfortunately I lost an example unit test which uses this function.

Reply:
I thing that another way to address this umr would be to change the serialize function to:

```
	template<typename Stream>
	void serialize(Stream& stream,ValueType& type)
	{
            if(Stream::isInput) {
                serializeVarInt7(stream,encodedValueType);
                type = (ValueType)-encodedValueType;
            } else {
		I8 encodedValueType = -(I8)type;
		serializeVarInt7(stream,encodedValueType);
            }
	}
```

Reply:
Yes, it seems both are good solutions.

Reply:
I slightly prefer my solution and I'll say why. The `serialize` function can either output the parameter `type` to the stream, or read from the stream and set it. When reading from the stream, it makes no sense to access the variable `type` just before overwriting what it contains.

So I think it is better to remove the useless read, rather than initializing the memory, so that the useless read we do does not read uninitialized memory.

But this is me being a little of a perfectionist... so I guess not a big deal we you leave it as is. Thanks!